deepchecks/tabular/context.py

```diff
@@ -88,7 +90,9 @@ def __init__(self,
             if dataset is not None:
                 feature_df_list.append(dataset.features_columns)
                 if y_pred is None and y_proba is not None:
+                    validate_proba(y_proba, model_classes)
                     y_pred = np.argmax(y_proba, axis=-1)
+                    y_pred = np.array(model_classes)[y_pred]
```

Comment:
Nice

Reply:
also possible `np.argmax(y_proba, axis=-1).map(model_classes.__getitem__)`

Reply:
Mine is more elegant